backend/middleware/authenticateToken.js

```diff
@@ -1,33 +1,27 @@
-const jwt = require("jsonwebtoken")
-
+const jwt = require("jsonwebtoken");
 
 const authenticateToken = (req, res, next) => {
-
-    // Get the token from the auth header (where it would be stored)
-    const authHeader = req.headers["authentication"];
-    const token = authHeader.split(' ')[1]
-
-    console.log(authHeader);
-
-    // Verify that there is a token
-    if(!token) {
-        // Bad request
-        // Code subject to change
-        res.sendStatus(401);
+  // Get the token from the auth header (where it would be stored)
+  const authHeader = req.headers["authentication"];
+  const token = authHeader.split(" ")[1];
+
+  // Verify that there is a token
+  if (!token) {
+    // Bad request
+    // Code subject to change
+    res.sendStatus(401);
+  }
+
+  jwt.verify(token, process.env.JWT_SECRET, (err, id) => {
+    if (err) {
```

Comment:
I just tested this on my end and this if statement actually doesn't do anything. (i.e. I can pass in a token without the hash and it will work as is)
You'll want to use a `try catch` on the jwt.verify function to tryly determine if the token is valid.

Refer to this example:

![image](https://user-images.githubusercontent.com/24628243/159829971-7fafd1d8-4ec6-4803-b30c-5dfb6bb8e5b8.png)
https://www.npmjs.com/package/jsonwebtoken

Reply:
Just to keep note of: While changing this, you'll want to test to make sure that you aren't sending multiple status' back either

Reply:
Please thoroughly test this

Reply:
Fixed: Changes are reflected in recent commits